File: typescript/src/n0021_merge_two_sorted_lists.test.ts
import { expect } from "jsr:@std/expect";
import { merge } from "./n0021_merge_two_sorted_lists.ts";
import { List } from "./utils/linked_list.ts";

Deno.test("merge", () => {
  const a = new List([1, 2, 4]);
  const b = new List([1, 3, 4]);
  a.head = merge(a.head, b.head);
  expect(a.toArray()).toEqual([1, 1, 2, 3, 4, 4]);
});

File: rust/src/n0021_merge_two_sorted_lists.test.rs
use crate::n0021_merge_two_sorted_lists::*;
use crate::utils::linked_list::*;

#[test]
fn test_merge() {
    let mut a = List::new([1, 2, 4]);
    let b = List::new([1, 3, 4]);
    a.head = merge(a.head, b.head);
    assert_eq!(a.to_vec(), vec![1, 1, 2, 3, 4, 4]);
}
